pkg/gateway/operations/listobjects.go

```diff
@@ -393,3 +399,43 @@ func (controller *ListObjects) Handle(w http.ResponseWriter, req *http.Request,
 		_ = o.EncodeError(w, req, nil, gatewayerrors.Codes.ToAPIErr(gatewayerrors.ErrBadRequest))
 	}
 }
+
+func handleListMultipartUploads(w http.ResponseWriter, req *http.Request, o *RepoOperation) {
+	o.Incr("list_multipart_uploads", o.Principal, o.Repository.Name, "")
+	resp := &serde.ListMultipartUploadsOutput{
+		Bucket: o.Repository.Name,
+	}
+
+	mpuResp, err := o.BlockStore.ListMultipartUploads(req.Context(), block.ObjectPointer{
+		StorageNamespace: o.Repository.StorageNamespace,
+		IdentifierType:   block.IdentifierTypeRelative,
+	})
+
+	if err != nil {
+		o.Log(req).WithError(err).Error("list multipart uploads failed")
+		_ = o.EncodeError(w, req, err, gatewayerrors.Codes.ToAPIErr(gatewayerrors.ErrInternalError))
+		return
+	}
+
+	var uploads []serde.Upload
+	for _, upload := range mpuResp.Uploads {
+		if upload.UploadId == nil {
+			continue
+		}
+		mpu, err := o.MultipartTracker.Get(req.Context(), *upload.UploadId)
+		if err != nil {
+			if errors.Is(err, kv.ErrNotFound) {
+				continue
+			}
+			o.Log(req).WithError(err).Error("could not read multipart record")
+			_ = o.EncodeError(w, req, err, gatewayerrors.Codes.ToAPIErr(gatewayerrors.ErrInternalError))
+			return
+		}
+		uploads = append(uploads, serde.Upload{
+			Key:      mpu.Path,
+			UploadID: *upload.UploadId,
+		})
+	}
+	resp.Uploads = uploads
+	o.EncodeResponse(w, req, resp, http.StatusOK)
```

Comment:
@nopcoder נrought to my attention this one-
Since in `HandleCreateMultipartUpload` we update the `MultipartTracker` with all the upload IDs we have, we might spare the `ListMultipartUploads` request for the object store by implementing a new `List` method for the `MultipartTracker` that will return the uploaded parts.